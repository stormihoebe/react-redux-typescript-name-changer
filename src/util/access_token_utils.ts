export const isLoggedIn = (key: string  = "access-token"): boolean => {
    const result: string | null = localStorage.getItem(key)
    return result !== null
}

export const getAccessToken = (): string => {
    const accessToken = localStorage.getItem("access-token")
    return accessToken ? accessToken : ""
}

export const deleteAccessToken = (): void => {
    localStorage.removeItem("access-token")
}

export const setAccessToken = (token: string): void => {
     localStorage.setItem("access-token", token)
}
