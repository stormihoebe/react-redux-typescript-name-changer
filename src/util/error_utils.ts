export const parseError = (statusCode: number) => {
    switch (statusCode) {
            case 403:
                return "Username and Password not found"
            case 401:
                return "You are not logged in"
            default:
                return "A server error occurred"
        }
}