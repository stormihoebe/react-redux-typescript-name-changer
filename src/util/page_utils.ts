import { RouteComponentProps } from "react-router"

export const genericRouteProps = {} as RouteComponentProps<{}>

export function setPageTitle(title: string): void {
    document.title = `${title} | Your App`
}