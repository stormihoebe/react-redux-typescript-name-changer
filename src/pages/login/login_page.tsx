import * as React from "react"
import * as redux from "redux"
import { RouteComponentProps } from "react-router"
import { connect } from "react-redux"
import { Store } from "../../redux/reducers"
import { setPageTitle } from "../../util/page_utils"
import { Button } from "../../components/button/button"
import { Error } from "../../redux/core"
import * as actions from "../../redux/actions"

const styles = require("./login_page.module.css")

interface ReduxState {
    readonly error: Error
}

interface ReduxActions {
    readonly login: (username: string, password: string) => void
    readonly receiveError: (errorText: string) => void
    readonly clearError: () => void
}

interface LoginPageProps extends RouteComponentProps<{}> {
}

interface LoginState {
    readonly username: string,
    readonly password: string,
    readonly errorText: string,
}

interface InputEvent {
    readonly target: {
        readonly name: string,
        readonly value: string,
    }
}

type LoginProps = ReduxState & ReduxActions & LoginPageProps & RouteComponentProps<{}>

const mapStateToProps = (state: Store.All, ownProps: LoginPageProps): ReduxState => ({
    error: state.error,
})

const mapDispatchToProps = (dispatch: redux.Dispatch<Store.All>): ReduxActions => ({
    login: (username, password) => dispatch( actions.login(username, password) ),
    receiveError: (errorText: string) => dispatch( actions.receiveError(errorText) ),
    clearError: () => dispatch( actions.clearError()),
})

export class LoginPage extends React.Component<LoginProps, LoginState> {
    constructor(props: ReduxState & ReduxActions & LoginPageProps) {
        super(props)
        this.state = {
            username: "",
            password: "",
            errorText: "",
        }
    }

    componentWillMount(): void {
        setPageTitle("Log in")
    }

    componentWillReceiveProps(nextProps: ReduxState & ReduxActions & LoginPageProps): void {
        if (nextProps.error.text !== this.props.error.text) {
            this.setState({errorText: nextProps.error.text})
        }
    }

    login = () => {
        const { username, password } = this.state
        if (!this.isValidInput(username) || !this.isValidInput(password)) { return }

        // clear out old error before submitting
        this.props.clearError()
        this.props.login(username, password)
    }

    handleChange = (event: InputEvent) => {
        const key = event.target.name.toLowerCase()

        if (key) {
            const meta = {}
            meta[key] = event.target.value
            this.setState({...meta, errorText: ""})

        }
    }

    isValidInput = (input: string) => {
        if (input.length === 0) {
            this.setState({errorText: "Fields cannot be blank"})
            return false
        }

        return true
    }

    render(): JSX.Element {
        return (
            <div className={styles.page}>
                <div className={styles.bg}/>
                <div className={styles.box}>
                    <h3 className={styles.h3}>
                        Log in
                    </h3>

                    <div className={styles.error}>{this.state.errorText}</div>
                    <div className={styles.form}>
                        <label className={`pt-label ${styles.label}`}>
                            Email
                            <input
                                type="email"
                                placeholder="your@email.com"
                                className={`pt-input ${styles.input}`}
                                value={this.state.username}
                                onChange={this.handleChange}
                                name={"username"}
                                autoFocus={true}
                            />
                        </label>

                        <label className={`pt-label ${styles.label}`}>
                            Password
                            <input
                                type="password"
                                className={`pt-input ${styles.input}`}
                                placeholder="••••••••••••••••"
                                value={this.state.password}
                                onChange={this.handleChange}
                                name={"password"}
                            />
                        </label>

                        <Button
                            text="Log in"
                            className={`pt-intent-primary pt-large ${styles.button}`}
                            kind="primary"
                            onClick={this.login}
                        />
                    </div>

                </div>
                <a href="#">Forgot password?</a>
            </div>
        )
    }
}

// export default LoginPage
const ConnectedApp: React.ComponentClass<LoginPageProps>
    = connect(mapStateToProps, mapDispatchToProps)(LoginPage)
export default ConnectedApp