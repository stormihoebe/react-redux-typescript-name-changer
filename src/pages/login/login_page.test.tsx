import * as React from "react"
import * as ReactDOM from "react-dom"

import { LoginPage } from "./login_page"
import { create as renderer } from "react-test-renderer"
import { genericRouteProps } from "../../util/page_utils"

const actions = {
    login: (username: string, password: string) => ({}),
    error: {text: ""},
    receiveError: (errorText: string) => ({}),
    clearError: () => ({}),
}

const renderLogin = () => (
   <LoginPage
            login={actions.login}
            error={actions.error}
            receiveError={actions.receiveError}
            clearError={actions.clearError}
            {...genericRouteProps}
        />
)

it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(renderLogin(), div)
})

it("renders a snapshot", () => {
    const tree = renderer(renderLogin()).toJSON()
    expect(tree).toMatchSnapshot()
})