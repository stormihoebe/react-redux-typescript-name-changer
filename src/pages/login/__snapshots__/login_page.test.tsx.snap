exports[`test renders a snapshot 1`] = `
<div
  className="page">
  <div
    className="bg" />
  <div
    className="box">
    <h3
      className="h3">
      Log in
    </h3>
    <div
      className="error">
      
    </div>
    <div
      className="form">
      <label
        className="pt-label label">
        Email
        <input
          autoFocus={true}
          className="pt-input input"
          name="username"
          onChange={[Function]}
          placeholder="your@email.com"
          type="email"
          value="" />
      </label>
      <label
        className="pt-label label">
        Password
        <input
          className="pt-input input"
          name="password"
          onChange={[Function]}
          placeholder="••••••••••••••••"
          type="password"
          value="" />
      </label>
      <button
        className="pt-button button primary pt-intent-primary pt-large button"
        disabled={undefined}
        kind="primary"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button">
        <span>
          Log in
        </span>
      </button>
    </div>
  </div>
  <a
    href="#">
    Forgot password?
  </a>
</div>
`;
